pkg/jobs/ingeststopped/ingesting_checker.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package ingeststopped
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/jobs"
+	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
+	"github.com/cockroachdb/cockroach/pkg/sql"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/catpb"
+	"github.com/cockroachdb/cockroach/pkg/sql/execinfrapb"
+	"github.com/cockroachdb/cockroach/pkg/sql/physicalplan"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/sql/types"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/cockroach/pkg/util/retry"
+	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
+	"github.com/cockroachdb/cockroach/pkg/util/tracing"
+	"github.com/cockroachdb/errors"
+)
+
+// WaitForNoIngestingNodes contacts all nodes in the cluster and verifies that
+// they are not ingesting data for the given job ID, and continues to try to do
+// so for the specified duration until it succeeds or times out.
+func WaitForNoIngestingNodes(
+	ctx context.Context, execCtx sql.JobExecContext, job *jobs.Job, maxWait time.Duration,
+) error {
+	ctx, sp := tracing.ChildSpan(ctx, "ingeststopped")
```

Comment:
I think we've been trying to use function names in these spans when it lines up one-to-one like this.